### DISCUSSION TITLE
Integration reports everything as 'unknown' after upgrading to 2024.7.0 (beta)

### DISCUSSION THREAD

**Discussion by Ndrinta** [nathanmarlor/foxess_modbus#635] [Q&A] [answered]
Just a heads up. Not sure if the devs are aware but I've tried upgrading to 2024.7.0b02 and later to b04 and in both instances the integration wasn't functional anymore.
Maybe this has been resolved already but worth mentioning 🙂
Keep up the good work! 💪

**Comment by canton7** [comment]
Thanks for the heads up! Anything in the logs indicating what the problem is? What does the Connection Status entity say?

**Comment by FozzieUK** [comment]
confirmed, these are the logs from startup

2024-07-01 11:00:46.787 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up foxess_modbus platform for sensor
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 364, in _async_setup_platform
await asyncio.shield(awaitable)
File "/config/custom_components/foxess_modbus/sensor.py", line 26, in async_setup_entry
async_add_devices(create_entities(SensorEntity, controller))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/foxess_modbus/inverter_profiles.py", line 330, in create_entities
return inverter_connection_type_profile_from_config(controller.inverter_details).create_entities(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/foxess_modbus/inverter_profiles.py", line 95, in create_entities
entity = entity_factory.create_entity_if_supported(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/foxess_modbus/entities/modbus_integration_sensor.py", line 53, in create_entity_if_supported
return ModbusIntegrationSensor(
^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/foxess_modbus/entities/modbus_integration_sensor.py", line 84, in __init__
IntegrationSensor.__init__(
TypeError: IntegrationSensor.__init__() missing 1 required keyword-only argument: 'max_sub_interval'

**Reply by Ndrinta** [reply]
You beat me.

**Reply by canton7** [reply]
Argh, ok thanks! Irritatingly I won't be able to fix this for another week or so

**Reply by Ndrinta** [reply]
No need to! Take it easy, no-one's life depends on it, we'll simply delay updating. 🙂

**Comment by FozzieUK** [comment]
I'm running a slightly older version of the integration v1.11.0 but these are the lines it is referring to -

![111](https://github.com/nathanmarlor/foxess_modbus/assets/113460294/41d94fd2-2761-48d4-a408-f3777cd19628)

**Comment by bbrks** [comment]
If you've upgraded to 2024.7.0 and want to temporarily fix this integration without downgrading, you can:
1. Add `from datetime import timedelta` near the top of the file.
2. Add `max_sub_interval=timedelta(minutes=5),` to the `IntegrationSensor.__init__` as shown below.

Do it via the file editor and restart HA after you've done this.

![Screenshot 2024-07-04 at 09 57 24](https://github.com/nathanmarlor/foxess_modbus/assets/1525809/e71c7c4f-3dad-4d8e-8dac-c28ea420fcc4)
![Screenshot 2024-07-04 at 09 57 16](https://github.com/nathanmarlor/foxess_modbus/assets/1525809/1645c8b9-58de-49c6-ba2d-c5c3ede34d00)

**Reply by Ndrinta** [reply]
This before or after you update to 2024.7.0?

**Reply by FozzieUK** [reply]
That would work on 2024.7.x, sadly it won't on earlier versions.
Whilst the changes to the Riemann sum integration sensor make max_sub_interval an optional parameter the actual fix will have to be more subtle and either take into account core version, or require setting a minimum core version for all future releases.

**Reply by bbrks** [reply]
Edit: Outdated comment - see parent for updated info.

**Reply by Ndrinta** [reply]
I'm not too fussed by staying on 2024.6.X for the time being, the only thing I'm eager on trying of 2024.7.X is the new resize feature but it can wait a few days/week. And certainly not worth the trade of with potentially losing loads of data from the PV system.

**Reply by Twilek-de** [reply]
The problem is not waiting with an update. The problem is that there is almost no way before updating to check wether some custom integrations might break or not (it was 2 for me in this release). If you have a lot of custom stuff it is not feasible to check each and every repository wether there are issues or discussions. Especially with the break-neck speed of the release cycle of HA. Don´t get me wrong. I am not blaming anyone here, on the contrary I want to thank you for the quick repsonse to the problem and the awesome work you are doing with this integration (almost 20% of my automations depend on it). I was just wondering wether maybe something could be included in HACS that users can mark releases as working for each integration or something of that sort so that there is a quick way to check if clicking the update button is safe.

**Reply by canton7** [reply]
I think just staying a week or two behind the cutting edge release is a reasonably good way of staying stable! I tend to sit a month or two behind the latest.

**Reply by bbrks** [reply]
I apply this pretty religiously in my professional life. I've _never_ had a bad experience waiting for a follow-up `.1` or `.2` release :)

**Reply by AndaleR** [reply]
Thanks. I have added those lines and the errors seems to be fixed.

Need to wait some time to get data into the energy-dashboard - but it seems to work.

Need to type a note: Wait for the next HA-update and install it some weeks later. But it was the first time in my small setup, so nothing was wrong until yesterday.

**Reply by canton7** [reply]
> The `max_sub_interval` documentation seems inaccurate, in both that it's required and not optional, but also None is not a suitable default.

Interestingly, using `max_sub_interval=None` *should* work, reading the source code. It just reverts back to the old behaviour.

And, indeed, I'm seeing it working fine locally.

**Comment by canton7** [comment, marked as answer]
Fixed in 1.12.1.
